typegraph/typegraph/providers/prisma/type_generator.py

```diff
@@ -327,9 +327,50 @@ def extract_number_types(self, tpe: t.struct) -> t.struct:
         return t.struct(fields)
 
     def get_order_by_type(self, tpe: t.struct) -> t.struct:
-        term_node_value = t.enum(["asc", "desc"]).optional()
-        remap_struct = self.deep_map(tpe, lambda _: term_node_value).optional()
-        return t.array(remap_struct)
+        # TODO: global, named, only generated once per runtime
+        sort = t.enum(["asc", "desc"]).optional()
+        aggregate = t.struct(
+            {k: sort.optional() for k in ["_count", "_avg", "_sum", "_min", "_max"]}
+        ).optional()
+
+        # orderByNulls is a preview feature in prisma: see
+        # https://www.prisma.io/docs/concepts/components/prisma-client/filtering-and-sorting#sort-with-null-records-first-or-last
+        # sort_nulls = t.struct({"sort": sort, "nulls": t.enum(["first", "last"])})
+
+        def get_sorting_type(tpe: t.TypeNode):
+            tpe = resolve_proxy(tpe)
+
+            # optional = False
+            if isinstance(tpe, t.optional):
+                # optional = True
+                tpe = tpe.of
+
+            if isinstance(tpe, t.struct):
+                # relation
+                # TODO: eventual infinite recursion
+                return get_order_by(tpe).optional()
+
+            if isinstance(tpe, t.array):
+                nested_type = resolve_proxy(tpe.of)
+                if isinstance(nested_type, t.struct):
+                    return aggregate
+                else:
+                    # TODO: check prisma docs
+                    return sort
+
+            # scalar type
+            # ? are they all sortable??
```

Comment:
I think the answer to this question would be yes (in most cases) since scalar types are either text or numbers (except arrays and BLOBs). But it can also depend on the target database. The Prisma engine should throw an error if it's not supported (in case the Typegraph allows it).